fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1096,7 +1137,8 @@ private void createJob(long dbId, OlapTable olapTable, Map<Long, LinkedList<Colu
                     for (Column alterColumn : alterSchema) {
                         if (alterColumn.nameEquals(partitionCol.getName(), true)) {
                             // 2.1 partition column cannot be modified
-                            if (!alterColumn.equals(partitionCol)) {
+                            if (!alterColumn.equals(partitionCol)
```

Comment:
checkPartitionColumnChangeAllowed() is the superset of Column.equals().
You can call Column.equals() in checkPartitionColumnChangeAllowed()
It can reduce some of the duplicate code paths.